hail/src/main/scala/is/hail/expr/ir/Pretty.scala

```diff
@@ -14,8 +15,8 @@ import org.json4s.jackson.{JsonMethods, Serialization}
 import scala.collection.mutable
 
 object Pretty {
-  def apply(ir: BaseIR, width: Int = 100, ribbonWidth: Int = 50, elideLiterals: Boolean = true, maxLen: Int = -1, allowUnboundRefs: Boolean = false): String = {
-    val useSSA = HailContext.getFlag("use_ssa_logs") != null
+  def apply(ctx: ExecuteContext, ir: BaseIR, width: Int = 100, ribbonWidth: Int = 50, elideLiterals: Boolean = true, maxLen: Int = -1, allowUnboundRefs: Boolean = false): String = {
+    val useSSA = ctx != null && ctx.getFlag("use_ssa_logs") != null
```

Comment:
There's one place where we use `null`: `MatrixBGENReaderParameters.toJValue`. You should interrogate that use. Maybe toJValue should be forced to take an ExecuteContext?

Reply:
I've always thought that usage was super weird -- a TableIR inside of a table reader parameters (and not treated as a child or anything). Passing null there is fine; we're going to fix it up at some point anyway. I think fixing this as part of the new functionality (TableReadWithFilter) intended to be added for fast web portal backends is the right strategy (this is there to support reading a BGEN with filter, the same functionality!)